distribution/lib/Standard/Base/0.0.0-dev/src/Runtime/Lazy_Ref.enso

```diff
@@ -0,0 +1,9 @@
+## PRIVATE
+   A simple type encapsulating the pattern of delaying initialization of some
+   value.
```

Comment:
Currently in Enso, I cannot create a lazy _value_. 

If you give me a way to initialize an inner value lazily, e.g.:
```
my_function group_builder x =
    ~lazy_val = expensive_computation x
    group_builder.specify "some test" <|
        lazy_val . should_equal 42
    group_builder.specify "some other test" <|
        lazy_val . should_be_a Integer
```
than I will not need `Lazy_Ref`.

Until then, every time I need a pattern like the above, I need to re-implement `Lazy_Ref`. I wanted to have this common functionality so that I can just use it, instead of having to copy-paste it every time I need it.

But if you are really strongly opposed to it I will just keep wasting time copy-pasting.

Reply:
I have removed the helper as per your request and I'm just adding the `type Lazy_Ref` wherever it is used.
Please approve the PR, so that I can proceed.